### PR TITLE
CDAP-6912 Instantiate application and configure it under impersonation

### PR DESCRIPTION
**Pull request by anwar6953**
Instantiate application and configure it under impersonation.
This is so that user can not easily perform operations as the cdap system user, via user code.

https://issues.cask.co/browse/CDAP-6912
http://builds.cask.co/browse/CDAP-RUT28-4

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java [hunk 1]
```diff
@@ -102,7 +106,12 @@ public InMemoryConfigurator(CConfiguration cConf, Id.Namespace appNamespace, Id.
     SettableFuture<ConfigResponse> result = SettableFuture.create();
 
     try {
-      Object appMain = artifactClassLoader.loadClass(appClassName).newInstance();
+      Object appMain = impersonator.doAs(artifactId.getNamespace().toEntityId(), new Callable<Object>() {
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java]: nit: use `appNamespace.toEntityId()`?

**Comment by bdmogal** [comment]
@anwar6953 code itself LGTM. But is it possible to add a unit test for this? Try to access something as cdap via a `configure()` method and fail?

**Comment by anwar6953** [comment]
Filed a JIRA for unit tests, which is a much larger task:
https://issues.cask.co/browse/CDAP-7070

**Comment by bdmogal** [comment]
LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java (hunk 1 above)
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java]: done; weird that we maintain namespace as well as artifactId.
@albertshau any idea why that's necessary?

**Comment by bdmogal** [comment]
LGTM

**Pull request merged**